modules/Emails/EmailsController.php

```diff
@@ -446,23 +446,45 @@ public function action_GetFromFields()
         $oe = new OutboundEmail();
         if ($oe->isAllowUserAccessToSystemDefaultOutbound()) {
             $system = $oe->getSystemMailerSettings();
-            $data[] = array(
-                'type' => 'system',
-                'id' => $system->id,
-                'attributes' => array(
-                    'from' => $system->mail_smtpuser,
-                    'name' => $system->name,
-                    'oe' => $system->mail_smtpuser,
-                ),
-                'prepend' => false,
-                'isPersonalEmailAccount' => false,
-                'isGroupEmailAccount' => true,
-                'outboundEmail' => array(
+
+            if($system->mail_smtpserver != 'smtp.sendgrid.net')
```

Comment:
The code seems to work fine, but I think might be done a bit cleaner. Mostly everything in this array is the same, instead of duplicating all this code consider declaring the entire array first, then modifying those keys that should have different values depending on the if statement.

Like:
```
$array = [
    'foo' => 1;
    ...
];

if ($sendgrid == 'bar') {
    $array['foo'] = 'baz';
}
```

I think it will result in a cleaner (and shorter) function.

Reply:
Yes this is a better approach.